src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+        uint256 _derived = (_balance * 40) / 100;
+        uint256 _adjusted = (((totalSupply() * DILL.balanceOf(account)) /
+            DILL.totalSupply()) * 60) / 100;
+        uint256 dillBoostedDerivedBal = Math.min(
+            _derived + _adjusted,
+            _balance
+        );
+
+        // Loop through the locked stakes, first by getting the liquidity * lock_multiplier portion
+        uint256 lockBoostedDerivedBal = 0;
+        for (uint256 i = 0; i < _lockedStakes[account].length; i++) {
+            LockedStake memory thisStake = _lockedStakes[account][i];
+            uint256 lock_multiplier = thisStake.lock_multiplier;
+            uint256 lastRewardClaimTime = _lastRewardClaimTime[account];
+            // If the lock is expired
+            if (
+                thisStake.ending_timestamp <= block.timestamp &&
+                !thisStake.isPermanentlyLocked
+            ) {
+                // If the lock expired in the time since the last claim, the weight needs to be proportionately averaged this time
+                if (lastRewardClaimTime < thisStake.ending_timestamp) {
+                    uint256 timeBeforeExpiry = thisStake.ending_timestamp -
+                        lastRewardClaimTime;
+                    uint256 timeAfterExpiry = block.timestamp -
+                        thisStake.ending_timestamp;
+
+                    // Get the weighted-average lock_multiplier
+                    uint256 numerator = (lock_multiplier * timeBeforeExpiry) +
+                        (_MultiplierPrecision * timeAfterExpiry);
+                    lock_multiplier =
+                        numerator /
+                        (timeBeforeExpiry + timeAfterExpiry);
+                }
+                // Otherwise, it needs to just be 1x
+                else {
+                    lock_multiplier = _MultiplierPrecision;
+                }
+            } else {
+                uint256 elapsedSeconds = block.timestamp - lastRewardClaimTime;
+                if (elapsedSeconds > 0) {
+                    lock_multiplier = thisStake.isPermanentlyLocked
+                        ? lockMaxMultiplier
+                        : _averageDecayedLockMultiplier(
+                            account,
+                            i,
+                            elapsedSeconds
+                        );
+                    _lastUsedMultiplier[account][i] =
+                        _lastUsedMultiplier[account][i] -
+                        (elapsedSeconds - 1) *
+                        multiplierDecayPerSecond;
+                }
+            }
+            uint256 liquidity = thisStake.liquidity;
+            uint256 combined_boosted_amount = (liquidity * lock_multiplier) /
+                _MultiplierPrecision;
+            lockBoostedDerivedBal =
+                lockBoostedDerivedBal +
+                combined_boosted_amount;
+        }
+
+        return dillBoostedDerivedBal + lockBoostedDerivedBal;
```

Comment:
`dillBoostedDerivedBal` reaches a maximum of the user's `_balance` to reflect a 2.5x boost whereas `lockBoostedDerivedBal` follows a different paradigm and reaches a maximum of `_balance * 2.5`, meaning that the impact of a locked stake outweighs that of a DILL lock. I believe they have to match up otherwise the math becomes incorrect.

Reply:
Good catch on the math difference between the 2 boosts. Need to rethink if this is possible similar to the DILL boost calculation. Although at the end the derived balance due to different boosts is relative (max being 2.5 and min being 1). I am thinking even if both boosts have different calcs, at the end it might be okay due to the relativity. I will still rethink this to maintain consistency.